templates-pipelines/windows-bios-installer/README.md

```diff
@@ -0,0 +1,100 @@
+
+## Prerequisites
+
+- KubeVirt `>=v1.0.0`
+- Tekton Pipelines `>=v0.44.0`
+
+### Obtain Windows ISO download URL
+
+1. Go to https://www.microsoft.com/en-us/software-download/windows10ISO.
+   You can also obtain a server edition for evaluation at https://www.microsoft.com/en-us/evalcenter/evaluate-windows-server-2019 (needs a different answer file!).
+2. Fill in the edition and `English` language (other languages need to be updated in windows10-autounattend ConfigMap) and go to the download page.
+3. Right-click on the 64-bit download button and copy the download link. The link should be valid for 24 hours.
+
+### Prepare autounattend.xml ConfigMap
+
+1. Supply, generate or use the default autounattend.xml.
+   For information on answer files see [Startup Scripts - KubeVirt User Guide](https://kubevirt.io/user-guide/virtual_machines/startup_scripts/#sysprep).
+2. Replace the default example autounattend.xml with your own in the definition of the `windows10-autounattend` ConfigMap in the pipeline YAML.
+   Different autounattend.xml can be also passed in a separate ConfigMap with the Pipeline parameter `autounattendConfigMapName` when creating a PipelineRun.
+
+## Pipeline Description
+
+```
+  create-vm-root-disk --- create-vm --- wait-for-vmi-status --- cleanup-vm
+```
+
+1. `create-vm-root-disk` task creates an empty DataVolume.
+2. `create-vm` task creates a VirtualMachine called `windows-bios-installer-*`
+   from the empty DataVolume and with the `windows-bios-installer-cd-rom` DataVolume attached as a CD-ROM.
+   A second DataVolume with the virtio-win ISO will also be attached (Pipeline parameter `virtioContainerDiskName`).
+3. `wait-for-vmi-status` task waits until the VirtualMachine shuts down.
+4. `cleanup-vm` deletes the installer VirtualMachine and ISO DataVolume (also in case of failure of the previous tasks).
+5. The output artifact will be the `win10` DataVolume with the basic Windows installation.
+   It will boot into the Windows OOBE and needs to be setup further before it can be used.
+
+## How to run
+
+Pipeline run with resolver:
+{% for item in pipeline_runs_yaml %}
+```yaml
+oc create -f - <<EOF
+{{ item | to_nice_yaml }}EOF
+```
+{% endfor %}
+
+### Usage in multiple namespaces
+
+When a user defines a different namespace in e.g. `baseDvNamespace`, then the serviceAccount under which the pipeline is running will require additional permissions in that namespace. Required permissions to run task in different namespace can be found in README of each task.
+
+## Possible Optimizations
+
+### Windows Source ISO Caching
+
+Windows source ISO can be downloaded and cached/stored in a DataVolume. So the subsequent PipelineRuns don't have to download the same ISO multiple times.
+
+You can for example create the following DataVolume first and remove create-source-dv step from the windows10-installer pipeline:
+
+```yaml
```

Comment:
Potentially put oc create around this too and define variable `WIN_IMAGE_DOWNLOAD_URL` beforehand.